### DISCUSSION TITLE
Problem in getting an SSL certificate with a caddy for a home server

### DISCUSSION THREAD

**Discussion by ParsaJR** [nextcloud/all-in-one#4930] [Questions] [answered]
Hello. I went through all the installation steps as per https://github.com/nextcloud/all-in-one/blob/main/local-instance.md .
It seems that the only problem for me is getting the certificate with Caddy (recommended reverse proxy method in https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md).
This error can be seen when i run Caddyfile 

> 2024/07/03 10:11:54.339 ERROR   http.acme_client        challenge failed        {"identifier": "cloud.parsajr.xyz", "challenge_type": "http-01", "problem": {"type": "urn:ietf:params:acme:error:dns", "title": "", "detail": "no valid A records found for cloud.parsajr.xyz; no valid AAAA records found for cloud.parsajr.xyz", "instance": "", "subproblems": []}}

> 2024/07/03 10:11:54.339 ERROR   http.acme_client        validating authorization        {"identifier": "cloud.parsajr.xyz", "problem": {"type": "urn:ietf:params:acme:error:dns", "title": "", "detail": "no valid A records found for cloud.parsajr.xyz; no valid AAAA records found for cloud.parsajr.xyz", "instance": "", "subproblems": []}, "order": "https://acme-v02.api.letsencrypt.org/acme/order/1815232447/283950116677", "attempt": 1, "max_attempts": 3}

It seems that the reason for this error is that **I pointed to a private ip in Cloudflare with my domain**! Therefore, this is not correct and the caddy cannot see it.
The question is, how should I solve this problem? How should Caddy see my home server?
In the first step, it is written that:
![image](https://github.com/nextcloud/all-in-one/assets/98831377/66536543-c399-4474-a3a8-1c4d99bef1ec)
Is it meant in this way?! Or should I point to my public ipv4? 👇
![image](https://github.com/nextcloud/all-in-one/assets/98831377/79bc5ae2-19cb-4c21-8337-ed36234296c4)

Also, in the second step, it is written that:
![image](https://github.com/nextcloud/all-in-one/assets/98831377/f0f07c71-0cef-4cc6-80ff-4e78b0cfc67c)
I mean, where exactly should I open port 80? My router? Should I open port 80 of the router and do port forwarding in order to complete acme challenge in Caddy?
sorry, maybe the question is awkward, but I'm a bit of a beginner in this field...

**Comment by szaimen** [comment, marked as answer]
> Or should I point to my public ipv4?

Yes

> My router? Should I open port 80 of the router and do port forwarding in order to complete acme challenge in Caddy?


Yes
